hugr-core/src/extension/prelude.rs

```diff
@@ -255,8 +247,92 @@ pub const ERROR_TYPE: Type = Type::new_extension(ERROR_CUSTOM_TYPE);
 pub const ERROR_TYPE_NAME: TypeName = TypeName::new_inline("error");
 
 /// Return a Sum type with the first variant as the given type and the second an Error.
-pub fn sum_with_error(ty: Type) -> SumType {
-    SumType::new([ty, ERROR_TYPE])
+pub fn sum_with_error(ty: impl Into<TypeRowRV>) -> SumType {
```

Comment:
would be good to use this in the existing code base, e.g. int ops return types (and possibly tests for values).

Reply:
This method was already there, and it _is_ used in multiple parts of the code.
I just made the signature sightly more general, to put it in line with the new helper functions.

Reply:
ah woops sorry